File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/collection/model/CollectionTranslations.java
package dev.reuss.tmdb.domain.collection.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.translation.Translation;
import dev.reuss.tmdb.common.translation.TranslationsResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Translations for a TMDB collection.
 *
 * @param id           collection id
 * @param translations collection translations
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record CollectionTranslations(
        int id,
        List<Translation<CollectionTranslationData>> translations
) implements TranslationsResponse<CollectionTranslationData> {

    public CollectionTranslations {
        translations = TmdbCollections.list(translations);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/collection/model/CollectionTranslations.kt
package dev.reuss.tmdb.domain.collection.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.translation.Translation
import dev.reuss.tmdb.common.translation.TranslationsResponse

/**
 * Translations for a TMDB collection.
 *
 * @property id collection id
 * @property translations collection translations
 */
@JvmRecord
data class CollectionTranslations(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("translations")
    override val translations: List<Translation<CollectionTranslationData>> = emptyList(),
) : TranslationsResponse<CollectionTranslationData>,
    TmdbModel
